web/components/answers/answer-resolve-panel.tsx

```diff
@@ -419,12 +422,49 @@ export const IndependentAnswersResolvePanel = (props: {
 
   const isAdmin = useAdmin()
   const user = useUser()
+  const answers = useAnswersCpmm(contract.id) ?? contract.answers
+  const isAdminorMod = useAdminOrMod()
+
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [error, setError] = useState<string | undefined>(undefined)
+
+  const onCancelAll = async () => {
+    setIsSubmitting(true)
+    try {
+      for (const answer of answers) {
+        const resolutionProps = removeUndefinedProps({
+          contractId: contract.id,
+          outcome: 'CANCEL',
+          answerId: answer.id,
+        })
+
+        try {
+          const result = await api(
+            'market/:contractId/resolve',
+            resolutionProps as any
+          )
+          console.log('resolved', resolutionProps, 'result:', result)
+        } catch (e) {
+          if (e instanceof APIError) {
+            setError(e.toString())
+          } else {
+            console.error(e)
+            setError('Error resolving question')
+          }
+        }
+      }
+      console.log('All contracts canceled')
+    } catch (e) {
+      console.error(e)
+      setError('Error canceling all contracts')
+    }
```

Comment:
You don't need the outer try catch if you have the inner one around each api call.